### ISSUE TITLE
Web [Windows] - Semantics announcement not announced by NVDA Screen Reader on Windows 

### ISSUE THREAD

**Issue by aakashawalkar** [flutter/flutter#122687]
## Steps to Reproduce

1. Execute `flutter run` on the code sample on Windows with a screen reader running (NVDA)
2. Enable the NVDA screen reader.
3. Click on the `Show SnackBar` button to display snackbar and then NVDA will not announce/read the text displayed in snackbar.

In the following code example, an announcement should be announced/read, but none is heard.

**Expected result:** The announcement is read by the screen reader.

**Actual result:** No announcement is read.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter SnackBar - NVDA Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MySnackBar(),
    );
  }
}

class MySnackBar extends StatelessWidget {
  const MySnackBar({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('SnackBar - NVDA Demo'),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.center,
          children: <Widget>[
            ElevatedButton(
              onPressed: () {
                /// Display SnackBar on button click
                ScaffoldMessenger.of(context).hideCurrentSnackBar();
                ScaffoldMessenger.of(context).showSnackBar(
                  const SnackBar(
                    duration: Duration(seconds: 2),
                    content: Text(
                      'Yay, I am Snack bar',
                    ),
                  ),
                );
              
                /// 
                SemanticsService.announce(
                  'I am a snackbar!',
                  TextDirection.ltr,
                );
              },
              child: const Text('Show SnackBar'),
            ),
          ],
        ),
      ),
    );
  }
}


```

</details>

<details>
  <summary>Logs</summary>

```
Analyzing accessibility-demo...                                         
No issues found! (ran in 2.0s)
```

```
Doctor summary (to see all details, run flutter doctor -v):

[✓] Flutter (Channel stable, 3.7.7, on Microsoft Windows [Version 10.0.22000.1574], locale en-US)
[✓] Windows Version (Installed version of Windows is version 10 or higher)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.76.0)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @aakashawalkar, I can reproduce the issue on the `stable` channel but not on the `master` channel. On the master channel the snackbar is announced as `I am a snackbar`. Can you confirm if this is the case for you as well?

<details>
<summary>flutter doctor of master channel</summary>

```
[!] Flutter (Channel master, 3.9.0-7.0.pre.33, on Microsoft Windows [Version 10.0.19045.2486], locale en-GH)
    • Flutter version 3.9.0-7.0.pre.33 on channel master at c:\src\sdks\master\flutter
    ! Warning: `flutter` on your path resolves to C:\src\sdks\stable\flutter\bin\flutter, which is not inside your   
      current Flutter SDK checkout at c:\src\sdks\master\flutter. Consider adding c:\src\sdks\master\flutter\bin to  
      the front of your path.
    ! Warning: `dart` on your path resolves to C:\src\sdks\stable\flutter\bin\dart, which is not inside your current 
      Flutter SDK checkout at c:\src\sdks\master\flutter. Consider adding c:\src\sdks\master\flutter\bin to the front      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb7e82826d (5 hours ago), 2023-03-15 04:34:21 -0400
    • Engine revision b7297f0259
    • Dart version 3.0.0 (build 3.0.0-322.0.dev)
    • DevTools version 2.22.2
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git"        
      directly to perform update checks and upgrades.

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at C:\Users\Deana\AppData\Local\Android\sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32526.322
    • Windows 10 SDK version 10.0.22000.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] IntelliJ IDEA Community Edition (version 2022.1)
    • IntelliJ at C:\Users\Deana\AppData\Local\JetBrains\Toolbox\apps\IDEA-C\ch-0\221.5921.22
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] IntelliJ IDEA Ultimate Edition (version 2022.1)
    • IntelliJ at C:\Users\Deana\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\221.5921.22
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.76.0)
    • VS Code at C:\Users\Deana\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.60.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19045.2486]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 110.0.5481.180
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 110.0.1587.63

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

**Comment by aakashawalkar** [comment]
Hello @danagbemava-nc - Yes, I see the issue is only on the `stable` channel but not on the `master` and `beta` [3.8.0-10.pre.1] channel.

**Comment by danagbemava-nc** [comment]
Thanks for the confirmation. 

Per our triage policy, I'll be closing this issue since it has already been fixed on the `beta`/`master` channels.
